### PR TITLE
dynamic subscription path

### PR DESCRIPTION
**Pull request by ImMohammad20000**

### CODE REVIEW

File: config.py [hunk 1]
```diff
@@ -29,6 +29,12 @@
 XRAY_ASSETS_PATH = config("XRAY_ASSETS_PATH", default="/usr/local/share/xray")
 XRAY_EXCLUDE_INBOUND_TAGS = config("XRAY_EXCLUDE_INBOUND_TAGS", default='').split()
 XRAY_SUBSCRIPTION_URL_PREFIX = config("XRAY_SUBSCRIPTION_URL_PREFIX", default="").strip("/")
+XRAY_SUBSCRIPTION_PATH = config("XRAY_SUBSCRIPTION_PATH", default="/sub/")
+
+if not XRAY_SUBSCRIPTION_PATH.startswith("/"):
```
Comment (M03ED) [inline config.py:34]: no need to add / in start , just add them in api

File: config.py [hunk 2]
```diff
@@ -29,6 +29,12 @@
 XRAY_ASSETS_PATH = config("XRAY_ASSETS_PATH", default="/usr/local/share/xray")
 XRAY_EXCLUDE_INBOUND_TAGS = config("XRAY_EXCLUDE_INBOUND_TAGS", default='').split()
 XRAY_SUBSCRIPTION_URL_PREFIX = config("XRAY_SUBSCRIPTION_URL_PREFIX", default="").strip("/")
+XRAY_SUBSCRIPTION_PATH = config("XRAY_SUBSCRIPTION_PATH", default="/sub/")
```
Comment (M03ED) [inline config.py:32]: receive XRAY_SUBSCRIPTION_PATH like this 
XRAY_SUBSCRIPTION_PATH = config("XRAY_SUBSCRIPTION_PATH", default="sub")

File: app/models/user.py [hunk 3]
```diff
@@ -277,7 +277,7 @@ def validate_links(cls, v, values, **kwargs):
     def validate_subscription_url(cls, v, values, **kwargs):
         if not v:
             token = create_subscription_token(values["username"])
-            return f"{XRAY_SUBSCRIPTION_URL_PREFIX}/sub/{token}"
+            return f"{XRAY_SUBSCRIPTION_URL_PREFIX}{XRAY_SUBSCRIPTION_PATH}{token}"
```
Comment (M03ED) [inline app/models/user.py:280]: return f"{XRAY_SUBSCRIPTION_URL_PREFIX}/{XRAY_SUBSCRIPTION_PATH}/{token}"

**Review by M03ED — CHANGES_REQUESTED** [review]
this can be better

**Comment by ImMohammad20000** [comment]
فکر نمیکنم /api مشکلی درست کنه🤔 باشه چک میکنم

**Comment by SaintShit** [comment]
there lots of file changes here, I can't check the real changes, I appreciate it if you make another PR for this

**Comment by ImMohammad20000** [comment]
Ok

**Pull request closed without merging**